client/client.go

```diff
@@ -281,21 +281,97 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
+func loadServerTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
+	// Trust CA's in the standard location (/etc/ssl/certs), and the
+	// configured server certificate when building the certificate chain
+	err := ctx.LoadVerifyLocations(conf.ServerCert, "/etc/ssl/certs/")
+	return ctx, err
+}
+
+func loadClientTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
+
+	if conf.HttpsClient == nil {
+		return ctx, errors.New("Empty HttpsClient config given")
+
+	}
+	certFile := conf.HttpsClient.Certificate
+	keyFile := conf.HttpsClient.Key
+
+	certBytes, err := ioutil.ReadFile(certFile)
+	if err != nil {
+		return ctx, errors.Wrap(err, "Failed to read the certificate file for the HttpsClient")
+	}
+
+	certs := openssl.SplitPEM(certBytes)
+	if len(certs) == 0 {
+		return ctx, fmt.Errorf("No PEM certificate found in '%s'", certFile)
+	}
+	first, certs := certs[0], certs[1:]
+	cert, err := openssl.LoadCertificateFromPEM(first)
+	if err != nil {
+		return ctx, err
+	}
+
+	err = ctx.UseCertificate(cert)
+	if err != nil {
+		return ctx, err
+	}
+
+	for _, pem := range certs {
+		cert, err := openssl.LoadCertificateFromPEM(pem)
+		if err != nil {
+			return ctx, err
+		}
+		err = ctx.AddChainCertificate(cert)
+		if err != nil {
+			return ctx, err
+		}
+	}
+
+	keyBytes, err := ioutil.ReadFile(keyFile)
+	if err != nil {
+		return ctx, errors.Wrap(err, "Private key file from the HttpsClient configuration not found")
+	}
+
+	key, err := openssl.LoadPrivateKeyFromPEM(keyBytes)
```

Comment:
just a comment: in here the potential private key loading from an Engine will take place (with EngineLoadPrivateKey).